llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -22714,32 +22714,7 @@ X86TargetLowering::BuildSDIVPow2(SDNode *N, const APInt &Divisor,
   if (Lg2 == 1)
```

Comment:
Just check the `Divisor == 2 || Division == -2`. The Lg2 made mode sense when we already needed Lg2 for the rest of the code, but that's in the TargetLowering::buildSDIVPow2WithCMov now.